app/Http/Controllers/AlbumsController.php

```diff
@@ -57,4 +59,90 @@ public function get()
 
 		return $return;
 	}
+
+	/**
+	 * @return array|string returns an array of photos of all albums or false on failure
+	 */
+	public function getPositionData()
+	{
+		// caching to avoid further request
+		Configs::get();
+
+		// Initialize return var
+		$return = array();
+		$return['albums'] = array();
+
+		// Get photos
+		// Get album information
+		$UserId = $this->sessionFunctions->id();
+		$full_photo = Configs::get_value('full_photo', '1') == '1';
+
+		$toplevel = $this->albumFunctions->getToplevelAlbums();
+		if ($toplevel === null) {
+			return Response::error('I could not find you.');
+		}
+
+		// All normal albums
+		$toplevel_albums = $this->albumFunctions->prepare_albums($toplevel['albums']);
+		// Add albums shared with users
+		$toplevel_albums = array_merge($toplevel_albums, $this->albumFunctions->prepare_albums($toplevel['shared_albums']));
+
+		$toplevel_album_ids = array();
+		$album_list = array();
+
+		foreach ($toplevel_albums as $album_iterator) {
+			$toplevel_album_ids[] = $album_iterator['id'];
+
+			$album_tmp = Album::with('children')->find($album_iterator['id']);
+			$full_photo = $album_tmp->full_photo_visible();
+
+			if ($album_tmp === null) {
+				Logs::error(__METHOD__, __LINE__, 'Could not find specified album');
+
+				return 'false';
+			}
+
+			// we just require is_logged_in for this one.
+			$username = null;
+			if ($this->sessionFunctions->is_logged_in()) {
+				$username = $album_iterator['owner'];
+			}
+
+			// Get all photos of subalbums -> you only call this function
+			// if you want all photos of subalbums
+			$album_list = array_merge($album_list, $this->albumFunctions->getAlbumIDsfromAlbumTree($this->albumFunctions->get_albums($album_tmp, $username)));
+		}
+
+		$album_list = array_merge($album_list, $toplevel_album_ids);
+		$photos_sql = Photo::whereIn('album_id', $album_list);
+
+		// Unsorted photos only visible for admin (album_id = null)
```

Comment:
Erm, no? Every user with upload rights can have unsorted photos.

See https://github.com/LycheeOrg/Lychee-Laravel/blob/master/app/Http/Controllers/SearchController.php#L193-L210 for an example of how to do this.